server/public/model/example_channel_test.go

```diff
@@ -0,0 +1,492 @@
+	client.SetToken(os.Getenv("MM_AUTHTOKEN"))
+
+	teamId := "team_id"
+	searchTerm := "search"
+	channels, _, err := client.SearchChannels(context.Background(), teamId, &model.ChannelSearch{
+		Term: searchTerm,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Found %d channels on team %s matching term '%s'\n", len(channels), teamId, searchTerm)
+}
+
+func ExampleClient4_SearchArchivedChannels() {
+	client := model.NewAPIv4Client(os.Getenv("MM_SERVICESETTINGS_SITEURL"))
+	client.SetToken(os.Getenv("MM_AUTHTOKEN"))
+
+	teamId := "team_id"
+	searchTerm := "search"
+	channels, _, err := client.SearchArchivedChannels(context.Background(), teamId, &model.ChannelSearch{
+		Term: searchTerm,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Found %d archived channels on team %s matching term '%s'\n", len(channels), teamId, searchTerm)
+}
+
+func ExampleClient4_GetChannelByName() {
+	client := model.NewAPIv4Client(os.Getenv("MM_SERVICESETTINGS_SITEURL"))
+	client.SetToken(os.Getenv("MM_AUTHTOKEN"))
+
+	channelName := "channel_name"
+	teamId := "team_id"
+	etag := ""
+	channel, _, err := client.GetChannelByName(context.Background(), channelName, teamId, etag)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Found channel %s with name %s\n", channel.Id, channel.Name)
+}
+
+func ExampleClient4_GetChannelByNameForTeamName() {
+	client := model.NewAPIv4Client(os.Getenv("MM_SERVICESETTINGS_SITEURL"))
+	client.SetToken(os.Getenv("MM_AUTHTOKEN"))
+
+	channelName := "channel_name"
+	teamName := "team_name"
+	etag := ""
+	channel, _, err := client.GetChannelByNameForTeamName(context.Background(), channelName, teamName, etag)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Found channel %s with name %s\n", channel.Id, channel.Name)
+}
+
+func ExampleClient4_GetChannelMembers() {
+	client := model.NewAPIv4Client(os.Getenv("MM_SERVICESETTINGS_SITEURL"))
+	client.SetToken(os.Getenv("MM_AUTHTOKEN"))
+
+	channelId := "channel_id"
+	page := 0
+	perPage := 60
+	etag := ""
+	members, _, err := client.GetChannelMembers(context.Background(), channelId, page, perPage, etag)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Found %d channel members for channel %s\n", len(members), channelId)
+}
+
+func ExampleClient4_AddChannelMember() {
+	client := model.NewAPIv4Client(os.Getenv("MM_SERVICESETTINGS_SITEURL"))
+	client.SetToken(os.Getenv("MM_AUTHTOKEN"))
+
+	channelId := "channel_id"
+	userId := "user_id"
+	cm, _, err := client.AddChannelMember(context.Background(), channelId, userId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Added user %s to channel %s with roles %s\n", userId, channelId, cm.Roles)
+
+	postRootId := "post_root_id"
+	cm, _, err = client.AddChannelMemberWithRootId(context.Background(), channelId, userId, postRootId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Added user %s to channel %s with roles %s using post %s\n", userId, channelId, cm.Roles, postRootId)
```

Comment:
Should this be its own example?

Reply:
This second example is specially in that it has a separate `Client4` API for what amounts to the same endpoint. Usually there's just one method with optional parameters.

But in either case, you're probably right that we want to support multiple examples for a single `OperationId` (using the OpenAPI terminology). Right now, it's a 1:1, but I'd be happy to revisit that as we flesh out more examples.

Reply:
Aaah, I understand the constraints now, it's one example per endpoint. It's good as is, then, it makes sense.